lib/facilities/gis_client.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require 'common/client/base'
+
+module Facilities
+  # Core class responsible for api interface operations
+  class GisClient < Common::Client::Base
+    configuration Facilities::GisConfiguration
+
+    def get_all_facilities(facility_type, order_field, max_record_count)
```

Comment:
Can we get some specs for this?
We should test edge cases for infinite loops, or unexpected results.
If `max_record_count` is records `per_page`, right?
And `query_count` is essentially a `page`, right?
Here's a scenario:
`max_record_count` is 10
`query_count` is 2 
total number of items in the external service is 20
Therefore we have all the records back but the `break` conditional has not been met
What happens when the loop runs again when `query_count` is 3?
What does that do to `data_collector`?  It's hard to picture because I don't know what `response.body` returns.